ckanext/canada/view.py

```diff
@@ -1200,3 +1201,61 @@ def members(id):
         u"group_type": 'organization'
     }
     return render(u'organization/members.html', extra_vars)
+
+
+def _promote_or_demote_sysadmin(username_or_id, sysadmin):
+    """
+    Promotes or demotes a user to/from a sysadmin.
+
+    Only sysadmins that are members of TBS can do this.
+    """
+    user = model.User.get(username_or_id)
+
+    if not user:
+        return abort(404, _('User not found'))
+
+    if user.name == g.user:
+        return abort(403, _('Cannot modify your own sysadmin privileges'))
+
+    tbs_membership = (model.Session.query(model.Group).
+                        filter(and_(model.Group.type == 'organization',
+                                    model.Group.state == 'active',
+                                    model.Group.name == 'tbs-sct')).
+                        join(model.Member, model.Member.group_id == model.Group.id).
+                        filter(and_(model.Member.table_id == g.userobj.id,
+                                    model.Member.table_name == 'user',
+                                    model.Member.state == 'active')))
+
+    # have to do mini-auth check here as sysadmins skip auth checks.
+    # we only want sysadmins that are part of TBS org and have TBS email.
+    # a bit crude, but don't want to be lax on sysadmin controllers.
+    if not is_sysadmin(g.user) or not tbs_membership or '@tbs-sct.gc.ca' not in user.email:
+        return abort(403, _('User %s not authorized to modify sysadmins.') % g.user)
+
+    try:
+        user.sysadmin = sysadmin
+        model.Session.add(user)
+        model.repo.commit_and_remove()
+        if sysadmin:
+            log.info('%s promoted %s to a sysadmin', g.user, user.name)
+            h.flash_success(_('Promoted %s to a sysadmin') % user.name)
+        else:
+            log.info('%s demoted %s from a sysadmin', g.user, user.name)
+            h.flash_success(_('Demoted %s from a sysadmin') % user.name)
+    except Exception:
```

Comment:
Can you catch something specific here? Otherwise might be better to raise the error so it doesn't get hidden from the logs

Reply:
@wardi I went into Core code and SQLalchemy to see what the commit/remove would raise. Does not look like anything. And Core CKAN also does not try/catch any of these.

So should be fine without the try/catch